File: server/db/data-dir.ts
import { homedir } from "node:os";
import { join } from "node:path";

export function getDataDir(): string {
  return join(homedir(), ".workbench");
}

export function getDbPath(): string {
  return join(getDataDir(), "data.db");
}

export function getScrollbackDir(): string {
  return join(getDataDir(), "scrollback");
}

export function getScrollbackPreviousDir(): string {
  return join(getScrollbackDir(), "previous");
}

File: server-rs/src/db/data_dir.rs
use std::path::PathBuf;

pub fn get_data_dir() -> PathBuf {
    dirs::home_dir()
        .map(|h| h.join(".workbench"))
        .unwrap_or_else(|| PathBuf::from(".workbench"))
}

pub fn get_db_path() -> PathBuf {
    get_data_dir().join("data.db")
}

pub fn get_scrollback_dir() -> PathBuf {
    get_data_dir().join("scrollback")
}

pub fn get_scrollback_previous_dir() -> PathBuf {
    get_scrollback_dir().join("previous")
}
